crp_APL/test/src/testNode.cpp

```diff
@@ -93,43 +103,99 @@ float crp::apl::TestNode::getYawFromQuaternion(const geometry_msgs::msg::Quatern
+void crp::apl::TestNode::mapPath()
+{
+    float currentPathLength = 0;
+    m_outPathMsg.points.clear();
+    m_noEnoughPointsLeft = true;
+    for (int i=0; i<m_localPath_x.size(); i++)
+    {
+        if (m_localPath_x.at(i)>=0)
+        {
+            // add points to path from simulated path
+            tier4_planning_msgs::msg::PathPointWithLaneId currentPoint;
+            currentPoint.point.pose.position.x = m_localPath_x.at(i);
+            currentPoint.point.pose.position.y = m_localPath_y.at(i);
+            tf2::Quaternion quat;
+            quat.setRPY(0.0f, 0.0f, m_localPath_theta.at(i));
+            currentPoint.point.pose.orientation = tf2::toMsg(quat);
+
+            currentPoint.point.longitudinal_velocity_mps = m_maximumSpeed;
+
+            float ds;
+            if (i==0)
+            {
+                ds = std::pow(std::pow(m_localPath_x.at(i),2)+
+                    std::pow(m_localPath_y.at(i),2),0.5);
+            }
+            else
+            {
+                ds = std::pow(std::pow(m_localPath_x.at(i)-m_localPath_x.at(i-1),2)+
+                    std::pow(m_localPath_y.at(i)-m_localPath_y.at(i-1),2),0.5);
+            }
+
+            currentPathLength = currentPathLength+ds;
+
+            m_outPathMsg.points.push_back(
+                currentPoint
+            );
+
+            if(std::abs(m_outPathMsg.points.at(0).point.pose.position.y > 3.0))
+            {
+                m_vehicleTooFarFromPath = true;
+            }
+            
+            if (currentPathLength > 100.0)
+            {
+                m_noEnoughPointsLeft = false;
+                break;
+            }
         }
     }
-    path.traffic_rules.maximum_speed = 20.0;
-    // only one lane at the moment
-    m_scenarioMsg.paths.push_back(path);
-    
+    printf("current path length %f\n", currentPathLength);
+    return;
+}
+
+void crp::apl::TestNode::mapObstacles()
+{
+    // to be filled up
+    return;
+}
+
+void crp::apl::TestNode::mapMovingObjects()
+{
+    // to be filled up
+    return;
+}
+
+void crp::apl::TestNode::mapDrivableSurface()
+{
+    // to be filled up
     return;
 }
 
 void crp::apl::TestNode::mapEgo()
 {
-    m_egoMsg.pose.pose.position.x = x[0];
-    m_egoMsg.pose.pose.position.y = x[1];
+    m_egoStatus.tire_angle_front = m_roadWheelAngleTarget;
+    m_kinematicState.pose_with_covariance.pose.position.x = x[0];
+    m_kinematicState.pose_with_covariance.pose.position.y = x[1];
     tf2::Quaternion(tf2::Vector3(0, 0, 1), x[2]);
-    m_egoMsg.pose.pose.orientation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0, 0, 1), x[2]));
-
-    m_egoMsg.twist.twist.linear.x = m_vehicleSpeedTarget;
+    m_kinematicState.pose_with_covariance.pose.orientation = 
+        tf2::toMsg(tf2::Quaternion(tf2::Vector3(0, 0, 1), x[2])); 
 
-    m_egoMsg.tire_angle_front = m_roadWheelAngleTarget;
+    m_kinematicState.twist_with_covariance.twist.linear.x = m_vehicleSpeedTarget;
 
     return;

```

Comment:
Maybe lateral acceleration can be publiseh too, or would be neat to have it in the future, for other types of controllers

Reply:
done